packages/flutter/test/material/tooltip_test.dart

```diff
@@ -14,20 +14,22 @@ import '../rendering/mock_canvas.dart';
 import '../widgets/semantics_tester.dart';
 import 'feedback_tester.dart';
 
-// This file uses "as dynamic" in a few places to defeat the static
-// analysis. In general you want to avoid using this style in your
-// code, as it will cause the analyzer to be unable to help you catch
-// errors.
-//
-// In this case, we do it because we are trying to call internal
-// methods of the tooltip code in order to test it. Normally, the
-// state of a tooltip is a private class, but by using a GlobalKey we
-// can get a handle to that object and by using "as dynamic" we can
-// bypass the analyzer's type checks and call methods that we aren't
-// supposed to be able to know about.
-//
-// It's ok to do this in tests, but you really don't want to do it in
-// production code.
+void _ensureTooltipVisible(GlobalKey key) {
+  // This function uses "as dynamic"to defeat the static analysis. In general
```

Comment:
Interesting that we have this big warning here, but not in all the other test files that do this.

Reply:
Yeah, this was something Ian asked for in the review when I was creating this test, before the pattern was established in the repo. He (rightly) was nervous about adding dynamic casts like this, and wanted me to make it clear that it wasn't for production code. I decided to preserve the warning, because it's a good warning. I thought about putting it elsewhere too, but it felt a bit like littering.